src/components/Proposal/ActionButton.tsx

```diff
@@ -68,12 +68,16 @@ class ActionButton extends React.Component<IProps, IState> {
   }
 
   public async handleClickExecute(_event: any): Promise<void> {
-    if (!(await enableWeb3ProviderAndWarn(this.props.showNotification.bind(this)))) { return; }
+    if (!await loadCachedWeb3Provider(this.props.showNotification) &&
```

Comment:
if we always use this pattern then shouldn't we have a function that calls and checks both loadCached and enable?

Reply:
yeah, I keep thinking the same thing.  I'll do it.

Reply:
Done